kubernetes-cronjob/massdriver.yaml

```diff
@@ -5,71 +5,191 @@ source_url: github.com/YOUR_ORG/<md .Name md>
 access: private
 type: application
 
+app:
+  envs:
+    # Use jq expressions to build environment variables from input params or connections
+    LOG_LEVEL: .params.configuration.log_level
+    # MONGO_DSN: .connections.mongo_authentication.data.authentication.username + ":" + .connections.mongo_authentication.data.authentication.password + "@" + .connections.mongo_authentication.data.authentication.hostname + ":" + (.connections.mongo_authentication.data.authentication.port|tostring)
+  policies: []
+    # Use jq expressions to select policies from artifact security blocks
+    # - .connections.sqs.data.security.iam.subscribe
+
 params:
+  required:
+    - namespace
+    - image
+    - resourceRequests
+    - job
   properties:
-    name:
-      type: string
     namespace:
-      type: string
+      title: Kubernetes Namespace
+      description: Application will be deployed into this namespace. If the namespace doesn't exist, it will be created.
+      $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/k8s-resource-name.json
       default: default
     image:
+      title: Container Image
       type: object
+      required:
+        - repository
+        - tag
       properties:
         repository:
-          type: string
-        sha:
+          $md.immutable: true
+          title: Repository
+          # Container repositories can be created or imported into massdriver at: https://app.massdriver.cloud/container-repositories
+          # You'll generally want to hard code your container repository in your bundle code, but it can be helpful to show it
+          # in your schema as well.
+          # If you need to allow an end-user developer to select the container repo, you can use our drop down widget documented here: https://docs.massdriver.cloud/bundles/custom-widgets-and-fields
+          description: Container repository to use for this application.
           type: string
         tag:
-          type: string
-    resource_requests:
+          title: Image Tag
+          $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/docker-image-tag.json
+          default: latest
+    command:
+      title: Command
+      description: Command to execute as the entrypoint of the container. If left unspecified, the container default will be used (referred to as 'entrypoint' in some container runtimes, such as Docker). Formatted as an array of strings so that 'exec' form is used.
+      type: array
+      items:
+        type: string
```

Comment:
These are so confusing for non-k8s people. Is it possible for us to take a string and split it into an array (quotes would be a not fun)

Reply:
Yeah, I thought about the same thing. There's at least one difficult problem to solve:

`bash -c tail -f /dev/null`

This `bash -c` pattern is very common in Dockerfiles and/or run commands - it tells bash to execute everything after the `-c` as the command, which makes the entire `tail -f /dev/null` is one argument.